### PR TITLE
change password

### PR DESCRIPTION
**Pull request by jikui**

### CODE REVIEW

File: bigip/common.sh [hunk 1]
```diff
@@ -106,12 +108,13 @@ function ha_build_cluster() {
...
     local selfhost=$2
     local peerhost=$3
+    local adminpass=$4
     
     echo "Build the HA cluster between $2 and $3"
 
-    code=$(curl -sk -u admin:admin -X POST  \
+    code=$(curl -sk -u admin:${adminpass} -X POST  \
         -H "Content-type: application/json" \
-        -d "{\"name\":\"DeviceGroup1\",\"type\":\"sync-failover\",\"autoSync\":\"enabled\",\"devices\":[\"$selfhost\",\"$peerhost\"]}" \
```
Comment (zhaoqin-github) [inline bigip/common.sh]: need to discuss autoSync enable/disable

File: bigip/common.sh [hunk 2]
```diff
@@ -165,25 +172,34 @@ function ha_failover()
...
 	if [ $? -eq 0 ]; then
 	   sleep 30
-	   ha_configure_sync $hostip
+	   ha_configure_sync $hostip  ${adminpass}
 	fi
     fi
     return 0
 }
 
+load_env_file()
```
Comment (zhaoqin-github) [inline bigip/common.sh]: do not recommend to load env file

File: bigip/build_ha_cluster.sh [hunk 3]
```diff
@@ -107,7 +107,10 @@ HOST_IDX=${HOST_IDX:-1}
 CLUSTER_NUM=${CLUSTER_NUM:-1}
 ALL_HOST=true
 
-while getopts i:n:h o ; do
+load_env_file
```
Comment (zhaoqin-github) [inline bigip/build_ha_cluster.sh]: do not recommend to load env file

File: bigip/build_ha_cluster.sh [hunk 4]
```diff
@@ -76,18 +76,18 @@ create_bigip_ha_cluster(){
         echo -e "\n$hostname $hostip $peerhostname $peerhostip"
         res=`expr $i % 2`
         if [ "$res" = "0" ]; then
-          build_ha_cluster $hostname $hostip $peerhostname $peerhostip
+          build_ha_cluster $hostname $hostip $peerhostname $peerhostip ${BIGIP_ADM_PASS}
```
Comment (zhaoqin-github) [inline bigip/build_ha_cluster.sh]: Not quite sure if we do need this password parameter, if that var is already exported

File: bigip/delete_bigip.sh [hunk 5]
```diff
@@ -135,18 +138,22 @@ delete_bigips() {
...
 
+load_env_file
 BIGIP_IDX=${BIGIP_IDX:-1}
 BIGIP_NUM=${BIGIP_NUM:-1000}
 
 BIGIQ_HOST=${BIGIQ_HOST:-10.250.11.123}
 BIGIQ_USER=${BIGIQ_USER:-admin}
 BIGIQ_PASS=${BIGIQ_PASS:-f5default}
-BIGIQ_POOL=${BIGIQ_POOL:-bigip}
+BIGIQ_POOL=${BIGIQ_POOL:-bigipnew}
```
Comment (zhaoqin-github) [inline bigip/delete_bigip.sh]: do not change default pool name

File: bigip/delete_bigip.sh [hunk 6]
```diff
@@ -135,18 +138,22 @@ delete_bigips() {
 
 # Main program start
 
+load_env_file
```
Comment (zhaoqin-github) [inline bigip/delete_bigip.sh]: do not load env file

File: bigip/list_bigip.sh [hunk 7]
```diff
@@ -162,13 +166,16 @@ TIMEOUT=${TIMEOUT:-30}
 COLUMN=("host" "bigip" "state" "mgmt" "mac" "ping" "icontrol" "license")
 COLUMN_WIDTH=($(cal_width ${COLUMN[@]}))
 
+load_env_file
```
Comment (zhaoqin-github) [inline bigip/list_bigip.sh]: do not load env file

File: bigip/onboard_bigip.sh [hunk 8]
```diff
@@ -44,6 +44,18 @@ usage() {
...
     -z \033[4mtime_zone\033[0m
         Specify time zone string for BIG-IP.
+    
+    -a \033[4bigip_admin_password\033[0m
+        Specify admin password for BIG-IP.
+     	
+    -b \033[4bigip_admin_new_password\033[0m
+        Specify new admin password for BIG-IP.
+
+    -r \033[4bigip_root_password\033[0m
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: -r will will duplicated the revoke parameter in relicense branch

File: bigip/onboard_bigip.sh [hunk 9]
```diff
@@ -64,14 +76,16 @@ usage() {
     BIGIP_DNS     DNS server ip address for BIG-IP.
     BIGIP_NTP     NTP server ip address for BIG-IP.
     BIGIP_TZ      Time zone string for BIG-IP.
+    BIGIP_ADM_PASS         BIGIP admin password
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: Maybe BIGIP_ADMIN_PASS will be better. ADM looks like a new module name of BIGIP.

File: bigip/onboard_bigip.sh [hunk 10]
```diff
@@ -192,15 +209,38 @@ onboard_bigips() {
       onboard $MGMT_IP
     fi
   done
+  save_new_env
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: do not do this

File: bigip/onboard_bigip.sh [hunk 11]
```diff
@@ -192,15 +209,38 @@ onboard_bigips() {
...
 }
 
+save_new_env()
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: Please remove it.

File: bigip/onboard_bigip.sh [hunk 12]
```diff
@@ -192,15 +209,38 @@ onboard_bigips() {
...
+	newline="export BIGIP_ROOT_PASS=$BIGIP_NEW_ROOT_PASS"
+	sed -i "$ a $newline" $envfile
+    fi
+}
 # Main program start
 
 BIGIP_IDX=${BIGIP_IDX:-1}
 BIGIP_NUM=${BIGIP_NUM:-1000}
 
+load_env_file
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: remove it

File: bigip/onboard_bigip.sh [hunk 13]
```diff
@@ -255,3 +318,5 @@ else
 
   echo -e "Complete onboarding BIG-IP in localhost"
 fi
+
+create_new_env_file
```
Comment (zhaoqin-github) [inline bigip/onboard_bigip.sh]: remove it

File: bigip/common.sh (hunk 1 above)
Comment (jikui) [reply bigip/common.sh]: Have already discussed with them and confirmed that they always set the value as the disabled.

File: bigip/onboard_bigip.sh (hunk 8 above)
Comment (jikui) [reply bigip/onboard_bigip.sh]: would you please use the other option for the relicense feature?  such as R?

**Pull request merged**